containers/agent/deploy.go

```diff
@@ -93,6 +191,111 @@ func ExecPodCommand(kclient kubernetes.Clientset, pod corev1.Pod, containerName
 	})
 }
 
+func allNodesToYaml(allNodesData string, filter bool) (string, error) {
+
+	allNodesUnstructured := make(map[string]interface{})
+	err := json.Unmarshal([]byte(allNodesData), &allNodesUnstructured)
+	if err != nil {
+		return "", err
+	}
+
+	if filter {
+		for _, val := range filteredAllNodesConfig {
+			delete(allNodesUnstructured, val)
+		}
+	}
+
+	// convert to string
+	yamlStr, err := yaml.Marshal(allNodesUnstructured)
+
+	return string(yamlStr), err
+}
+
+func getCtlplaneExports(dClient dynamic.Interface) (string, error) {
+
+	configVersion := dClient.Resource(openstackConfigVersionGVR)
+
+	unstructuredConfigVersion, err := configVersion.Namespace(deployOpts.namespace).Get(context.Background(), os.Getenv("CONFIG_VERSION"), metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%v", unstructuredConfigVersion.Object["spec"].(map[string]interface{})["ctlplaneExports"]), nil
+
+}
+
+func processOvercloudJSON(kclient kubernetes.Clientset, config *rest.Config) error {
+	var err error
+	glog.V(0).Info("Running process Overcloud JSON.")
+
+	// lookup the deployment pod (openstackclient)
+	pod, err := kclient.CoreV1().Pods(deployOpts.namespace).Get(
+		context.TODO(),
+		deployOpts.pod,
+		metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// obtain the allNodesData from the overcloud.json from the deployment pod
+	// this is written by TripleO's Ansible after successful deployment
+	allNodesDataString, err := CopyFromPod(kclient, *pod, "openstackclient", "/home/cloud-admin/work/"+deployOpts.configVersion+"/playbooks/tripleo-ansible/group_vars/overcloud.json")
```

Comment:
not required in this patch, but we might also want to get the clouds.yaml and /etc/hosts from the deployment. in case there is an issue with the storage of the openstackclient, thats some useful information to restore it.

Reply:
@stuggi yeah, maybe we file this as Jira tickets and can follow up on it as a "core" issue.